### ISSUE TITLE
How to write the enum have the defaut value ?

### ISSUE THREAD

**Issue by SteveMobile** [dropbox/djinni#250]
hi, We need write a enum, which has the defaut value
such as : 
enum DATA {
DATA1 = 2;
DATA2 = 3;
}

**Comment by artwyman** [comment]
There's no support for that in Djinni currently.  The best I can suggest with the existing generator would be to create a const map which can be used to lookup your enum values and get the corresponding integers.

Such a feature could be added, though concept doesn't translate perfectly into Java, where an enum's default ordinal can't be changed.  It is possible to give a Java enum a non-default "value" member variable, though.

**Comment by kennykaye** [comment]
We've also had the need to specify typed enums. Our workaround is the same as @artwyman indicated.

A couple of thoughts about the limitations of such a feature:

**TL;DR**: Limited support for integer typed enum support seems possible.

* C++11 [supports](http://en.cppreference.com/w/cpp/language/enum) typed enums, while earlier versions do not. There would need to be either some validation mechanism to not allow typed arrays for earlier versions, or a wrapper of sorts.
* Objective-C `NS_ENUM` appears to support integers but not other types.
* Java can support typed enums pretty easily, though accessing the value would require a getter:

```java
public enum Color {
    Unknown(-1),
    Red(1),
    Green(2),
    Blue(3);

    private final int mValue;

    Color(int value) {
        mValue = value;
    }

    public value getValue() {
        return mValue;
    }
}
```

**Comment by guycnicholas** [comment]
It was my understanding that djinni requires C++11, so I don't think it would be necessary to worry about earlier versions.

**Comment by artwyman** [comment]
Djinni does require C++11, and always uses typed enums.  I don't think any change to that is required for the request here.  I think the core of the request here is to allow specifying explicitly what the numerical value of an enum should be, rather than accepting the default/unspecified value (which will be assigned to 0...N in order).  Typed enums do have integer values, and can be cast to int.  They just don't cast implicitly, the way untyped enums do.
